src/Components/GameBoard.js

```diff
@@ -37,24 +41,38 @@ class GameBoard extends Component {
   constructor(props) {
     super(props);
     this.customNodeRef = React.createRef();
+    this.myRef = React.createRef();
 
     this.state = {
       graph: sample,
       selected: {},
       layoutEngineType: 'VerticalTree',
-
+      addModalClose:false,
+      addModalShow:false,
       loading: true,
       board: null,
       gameID: null,
       playerCardChoice: null,
-      playerBalanceAttempt: null
-
+      playerBalanceAttempt: null,
+      difficulty:null,
+      players:null,
+      data_structure:null
     };
   }
   // TODO:  ADD COMMENTS HERE
   async componentDidMount() {
+        const cookies = new Cookies();
```

Comment:
Create an issue for the next sprint to type check here before making an API call to ensure the user has given correct input.